sanitize_html/lib/src/html_formatter.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+
+import 'package:html/dom.dart';
+
+final _attrEscape = HtmlEscape(HtmlEscapeMode.attribute);
+final _textEscape = HtmlEscape(HtmlEscapeMode.unknown);
+
+String formatHtmlNode(Node node) {
+  return _HtmlFormatter()._format(node);
+}
+
+class _HtmlFormatter {
+  final _sb = StringBuffer();
+
+  String _format(Node node) {
+    _writeNodes([node], 0);
+    return _sb.toString();
+  }
+
+  void _writeNodes(List<Node> nodes, int level) {
+    if (nodes == null || nodes.isEmpty) return;
+    for (Node node in nodes) {
+      if (node is Element) {
+        _writeElement(node, level);
+      } else if (node is Text) {
+        _sb.write(_textEscape.convert(node.text));
+      } else if (node is Document) {
+        _writeNodes([node.documentElement], level);
+      } else if (node is DocumentFragment) {
+        _writeNodes(node.nodes, level);
+      } else {
+        throw UnimplementedError('Unknown node: ${node.runtimeType} ($node)');
+      }
+    }
+  }
+
+  void _writeElement(Element elem, int level) {
+    final tagName = elem.localName;
+    _sb.write('<$tagName');
```

Comment:
We are confident that `tagName` doesn't need escaping right? because `package:html` is an HTML5 parser...

Reply:
We could escape it like it were a text content... along with the attribute names?

Reply:
I think we care confident, but maybe we should try to trick the parser and see if it'll produce sketchy tag names...

Oh, never mind, of course this is sane... We're sanitizing all the tag names and only allowing whitelisted tagnames. so How could they possibly have a tagName that needs to be escaped :)